src/main/java/org/opentripplanner/transit/raptor/rangeraptor/multicriteria/arrivals/TransferStopArrival.java

```diff
@@ -10,6 +10,8 @@
  */
 public final class TransferStopArrival<T extends RaptorTripSchedule> extends AbstractStopArrival<T> {
 
+    private final RaptorTransfer transfer;
+
```

Comment:
I expected this change to degrade the performance, but I have not measured any change. I did run the Speed test for this PR. I also looked at who many new instance one travel search creates for this class, witch is 2*10^6 (max for the 27 cases i have in my benchmark test).